modules/engage-paella-player-7/src/js/EpisodeConversor.js

```diff
@@ -278,13 +290,15 @@ function processAttachments(episode, manifest, config) {
       });
     }
     else {
+      // TODO: will this cause an endless recursion loop when there are no
+      // navigation segments preview images within the mediapackage?
       videoPreview = getVideoPreview(episode.mediapackage, config);
```

Comment:
The performance could be improved:
```suggestion
      if (videoPreview === null) {
        videoPreview = getVideoPreview(episode.mediapackage, config);
      }
```

Reply:
I added the conditional around that call. Thanks